### PR TITLE
Add timeline events restoring

### PR DESCRIPTION
**Pull request by ranquild**
Epic: https://github.com/metabase/metabase/issues/20289
Demo: https://www.loom.com/share/4ec4135cc7c84273a5c63c743374bc21

Changes:
- Adds a modal that displays a list of archived events for a timeline.
- Adds a `queryKey` to all requests where an entity is loaded by id. When the `query` used to load an entity changes, the cached entity will be invalidated. This is needed to reload timelines when navigating between modals displaying archived and non-archived data.

How to test:
- Open a timeline with some events
- Use some event's menu to archive it
- Use the timeline's menu to go to archived timelines
- Use some event's menu to restore it
- Make sure you can navigate with browser buttons between the regular and archived timeline modals and events are displayed correctly

### CODE REVIEW

File: frontend/src/metabase/timelines/containers/TimelineModal/TimelineModal.tsx [hunk 1]
```diff
@@ -19,12 +19,16 @@ const collectionProps = {
     Urls.extractCollectionId(props.params.slug),
 };
 
+const mapStateToProps = () => ({
+  archived: false,
+});
+
```
Comment (kulyk) [inline frontend/src/metabase/timelines/containers/TimelineModal/TimelineModal.tsx:25]: Why do we need to hardcode `archived: false` here?

**Review by kulyk — APPROVED** [review]
Looks good! Just have a small question about `mapStateToProp`. Also, we use the word "unarchive" for all the other entities, but we use "restore" for events. Is this intentional?

**Comment by ranquild** [comment]
> Looks good! Just have a small question about `mapStateToProp`. Also, we use the word "unarchive" for all the other entities, but we use "restore" for events. Is this intentional?

No it's not, I'll fix that

File: frontend/src/metabase/timelines/containers/TimelineModal/TimelineModal.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/timelines/containers/TimelineModal/TimelineModal.tsx:26]: I've made `archived` prop optional in `TimelineModal` with default to `false`.

**Pull request merged**